chia/wallet/wallet_state_manager.py

```diff
@@ -802,8 +802,16 @@ async def new_coin_state(
             # if the new coin has not been spent (i.e not ephemeral)
             elif coin_state.created_height is not None and coin_state.spent_height is None:
                 await self.coin_added(
-                    coin_state.coin, coin_state.created_height, all_unconfirmed, wallet_id, wallet_type, coin_name
+                    coin_state.coin,
+                    coin_state.created_height,
+                    all_unconfirmed,
+                    wallet_id,
+                    wallet_type,
+                    coin_name,
+                    skip_lookup=local_record is None,
```

Comment:
it's not obvious to me why it's safe to skip this check here

Reply:
If local record is none, there is no reason to do the same exact lookup again. It will just be none, and we will continue